tools/distpartitioning/data_shuffle.py

```diff
@@ -794,23 +814,25 @@ def read_dataset(rank, world_size, id_lookup, params, schema_map):
         world_size,
         params.num_parts,
         schema_map,
+        ntype_counts,
     )
     logging.info(f"[Rank: {rank}] Done reading dataset {params.input_dir}")
+    dist.barrier()  # SYNCH
 
     edge_data = augment_edge_data(
         edge_data, id_lookup, edge_tids, rank, world_size, params.num_parts
     )
+    dist.barrier()  # SYNCH
```

Comment:
adding `dist.barrier()` before or after the calls looks a bit tedious. How about sync within the target call?

Reply:
As I mentioned earlier, I did for debugging purposes. 

We can do away with these barrier calls, since messaging calls themselves act as barriers, by definition. 

Also, Please note that we are targeting massively large datasets. We want to debug these datasets in as much less time as possible. In these instances, Since we ensure all the ranks run the major steps of the pipeline in a lock-step fashion, the debugging can be easier. 

I will leave it upto you. Lets chat about this in the call later this week.